src/coreclr/pal/src/arch/riscv64/context2.S

```diff
@@ -156,6 +156,9 @@ LOCAL_LABEL(Done_CONTEXT_CONTROL):
     and  t3, t1, t0
     bne  t3, t0, LOCAL_LABEL(Done_CONTEXT_INTEGER)
 
+    li  t0, 0
+    sd  t0, (CONTEXT_X0)(a0)
+
```

Comment:
My suggestion is 
```
     ld  t1, 8(sp)
     ld  t3, 16(sp)
 
+    sd  x0, (CONTEXT_X0)(a0)
     sd  tp, (CONTEXT_Tp)(a0)
     sd  gp, (CONTEXT_Gp)(a0)
     sd  a0, (CONTEXT_A0)(a0)
```
If I misunderstand your code, please let me know.
Thank you.

Reply:
Same logic but much better (for some reason I thought `sd` don't work with `x0` directly and don't even check it -_-). Thanks!